### DISCUSSION TITLE
Generating JSON file from program

### DISCUSSION THREAD

**Discussion by MarianBaba** [eclipse-langium/langium#1202] [Q&A] [answered]
Hello everyone!

While using the playground provided [here](https://langium.org/playground/), I noticed that there is a dinamically updated json form of the program's AST. I was wondering if there is a way to do this in VS Code with programs written using a DSL I designed. I know a json form of the grammar is generated automatically, but I would like to know if there is a way to do this for programs while they are being written using a custom designed DSL. 

For clarification, this is what I mean (right side of the image):
![Immagine 2023-09-27 135122](https://github.com/eclipse-langium/langium/assets/78113501/3d26f005-149c-4d17-ac8e-dd7fcf7cf837)

thank you in advance for the help!

**Comment by msujew** [comment, marked as answer]
@MarianBaba You can use the `JsonSerializer` service to generate such a JSON document. It is available in the [language injection container](https://github.com/eclipse-langium/langium/blob/2187f8355ee8fe18994d3c9779327ada3456af03/packages/langium/src/services.ts#L127-L129).

https://github.com/eclipse-langium/langium/blob/2187f8355ee8fe18994d3c9779327ada3456af03/packages/langium/src/serializer/json-serializer.ts#L64-L78

**Comment by cdietrich** [comment]
are you looking for something like  `console.log(services.serializer.JsonSerializer.serialize(model))`
